src/kokkos_ext/ArborX_KokkosExtSort.hpp

```diff
@@ -12,179 +12,12 @@
-  bin_sort.sort(space, values);
-}
-
-#if defined(KOKKOS_ENABLE_CUDA) ||                                             \
-    (defined(KOKKOS_ENABLE_HIP) && defined(ARBORX_ENABLE_ROCTHRUST))
-template <typename Keys, typename Values>
-void sortByKey(
-#if defined(KOKKOS_ENABLE_CUDA)
-    Kokkos::Cuda const &space,
-#else
-    Kokkos::HIP const &space,
-#endif
-    Keys &keys, Values &values)
-{
-  Kokkos::Profiling::ScopedRegion guard("ArborX::KokkosExt::sortByKey::Thrust");
-
-  using ExecutionSpace = std::decay_t<decltype(space)>;
-  static_assert(Kokkos::is_view<Keys>::value);
-  static_assert(Kokkos::is_view<Values>::value);
-  static_assert(Keys::rank() == 1);
-  static_assert(Values::rank() == 1);
-  static_assert(KokkosExt::is_accessible_from<typename Keys::memory_space,
-                                              ExecutionSpace>::value);
-  static_assert(KokkosExt::is_accessible_from<typename Values::memory_space,
-                                              ExecutionSpace>::value);
-  auto const n = keys.size();
-  ARBORX_ASSERT(values.size() == n);
-
-  if (n == 0)
-    return;
-
-#if defined(KOKKOS_ENABLE_CUDA)
-  auto const execution_policy = thrust::cuda::par.on(space.cuda_stream());
-#else
-  auto const execution_policy = thrust::hip::par.on(space.hip_stream());
-#endif
-
-  thrust::sort_by_key(execution_policy, keys.data(), keys.data() + n,
-                      values.data());
-}
-#endif
-
-#if defined(KOKKOS_ENABLE_SYCL) && defined(ARBORX_ENABLE_ONEDPL)
-template <typename Keys, typename Values>
-void sortByKey(Kokkos::Experimental::SYCL const &space, Keys &keys,
-               Values &values)
-{
-  Kokkos::Profiling::ScopedRegion guard("ArborX::KokkosExt::sortByKey::OneDPL");
-
-  using ExecutionSpace = std::decay_t<decltype(space)>;
-  static_assert(Kokkos::is_view<Keys>::value);
-  static_assert(Kokkos::is_view<Values>::value);
-  static_assert(Keys::rank() == 1);
-  static_assert(Values::rank() == 1);
-  static_assert(KokkosExt::is_accessible_from<typename Keys::memory_space,
-                                              ExecutionSpace>::value);
-  static_assert(KokkosExt::is_accessible_from<typename Values::memory_space,
-                                              ExecutionSpace>::value);
-  auto const n = keys.size();
-  ARBORX_ASSERT(values.size() == n);
-
-  if (n == 0)
-    return;
-
-  oneapi::dpl::execution::device_policy policy(
-      *space.impl_internal_space_instance()->m_queue);
-#if ONEDPL_VERSION_MAJOR > 2022 ||                                             \
-    (ONEDPL_VERSION_MAJOR == 2022 && ONEDPL_VERSION_MINOR >= 2)
-  oneapi::dpl::sort_by_key(policy, keys.data(), keys.data() + n, values.data());
-#else
-  auto zipped_begin =
-      oneapi::dpl::make_zip_iterator(keys.data(), values.data());
-  oneapi::dpl::sort(
-      policy, zipped_begin, zipped_begin + n,
-      [](auto lhs, auto rhs) { return std::get<0>(lhs) < std::get<0>(rhs); });
-#endif
+using Kokkos::Experimental::sort_by_key;
```

Comment:
Is there a good reason to keep the alias?

Reply:
Yes. This way, we contain the changes in this PR to a single file. Otherwise, we would have to do the changes all over twice: in this PR, and once `sort_by_key` moves out of `Experimental`.